pages/api/upload_to_drive.ts

```diff
@@ -9,41 +9,92 @@ import { supabase } from "../../utils/supabase";
  * */
 async function uploadFile(
   fileName: string,
+  folderName: string,
 ) {
 
-  const gdriveSettings = process.env.GOOGLE_DRIVE_SETTINGS || ""
-  const driveFolderId = process.env.GOOGLE_DRIVE_FOLDER_ID || ""
-  const driveCredentials = JSON.parse(gdriveSettings)
+  const gdriveSettings = process.env.GOOGLE_DRIVE_SETTINGS || "";
+  const driveFolderId = process.env.GOOGLE_DRIVE_FOLDER_ID || "";
+  const driveCredentials = JSON.parse(gdriveSettings);
   const scopes = ["https://www.googleapis.com/auth/drive"];
   const { data, error } = await supabase.storage.from("pdfs").download(fileName);
 
   if (error) throw error;
 
   if (!data) throw new Error("No data received from Supabase.");
 
-  const stream = await createStream(data)
+  const stream = await createStream(data);
 
   const auth = new google.auth.GoogleAuth({
     credentials: driveCredentials,
     scopes: scopes
   });
 
-  const driveService = google.drive({ version: 'v3', auth })
+  async function resolveParentFolderId(folderName: string): Promise<string> {
+    // Check if the folder already exists
+    const folderListResponse = await driveService.files.list({
+      q: `mimeType='application/vnd.google-apps.folder' and trashed=false and name='${folderName}'`,
+      fields: "nextPageToken, files(id, name)",
+      spaces: "drive",
+    });
+
+    const folderId = folderListResponse.data.files?.[0]?.id;
+    if (folderId) {
+      return folderId;
+    }
+
+    const folderMetadata = {
+      name: folderName,
+      mimeType: "application/vnd.google-apps.folder",
+      parents: [driveFolderId],
+    };
+
+    const folderResponse = await driveService.files.create({
+      requestBody: folderMetadata
+    });
+    return folderResponse.data.id!;
+  };
+
+  async function resolveFileNameForDrive(initialFileName: string): Promise<string> {
+    let currentFileName = initialFileName;
+    let finished = false;
```

Comment:
Can we rename this to something descriptive, like `fileWithout_iSuffixFound` or if you have more clear name, even better. I spend some time trying to figure out what that code in while does.

Adding comment above first if (where we are checking if `files` are found) would be beneficial for quicker understanding code.